src/core/protocol-authorization.ts

```diff
@@ -77,9 +77,18 @@ export class ProtocolAuthorization {
     incomingMessage: RecordsWrite,
     messageStore: MessageStore,
   ): Promise<void> {
-    // fetch ancestor message chain
-    const ancestorMessageChain: RecordsWriteMessage[] =
-      await ProtocolAuthorization.constructAncestorMessageChain(tenant, incomingMessage, incomingMessage, messageStore);
+    const existingInitialWrite = await ProtocolAuthorization.fetchInitialWrite(tenant, incomingMessage.message.recordId, messageStore);
+
+    let descendantRecordId;
+    if (existingInitialWrite === undefined) {
+      // NOTE: the incoming message "should" be an initial write at this point,
+      // but we don't check if it is here because it is checked in the RecordsWriteHandler already (even though at a later stage).
```

Comment:
@LiranCohen , you are spot on. This one is tough, if I don't say anything here, then people reading the condition `if (existingInitialWrite === undefined) {` could be wondering:

Hm... why do we assume that the write is an initial write just because an initial write doesn't already exist? What if this write is NOT an initial write, would looking at the `parentId` introduce a bug?

For this reason I actually attempted to reshuffle the logic in `RecordsWriteHandler` but backed out because it comes even more involved with a multiple tests failing. May not be too hard to fix, but decided to back out to get mini-wins iteratively.

I won't have the battle if you think we should remove entirely, even better if you can suggest better wording! Perhaps:

```suggestion
      // NOTE: we can assume this message is an initial write because 
      // we check in the `RecordsWriteHandler` and fail if the incoming message is not an initial write and no initial write exists (even though at a later stage).
```

Reply:
Not a hill I'll die on, but maybe adding more verbosity would help clear it up vs removing it completely?
Something along the lines of:

```suggestion
// NOTE: we can assume this message is an initial write because an existing initial write does not exist.
// Additionally, we check further down in the `RecordsWriteHandler` if the incoming message is an initialWrite, so we don't check explicitly here to avoid an unnecessary duplicate check.
```

Feel free to change the wording a bit, but I think the intent is a little more clear? It's not too important of a hill for me so I'll leave it to your discretion.

Reply:
Took you suggestion.